estuary-cdk/estuary_cdk/capture/common.py

```diff
@@ -611,14 +611,32 @@ async def _binding_incremental_task(
     connector_state = ConnectorState(
         bindingStateV1={binding.stateKey: ResourceState(inc=state)}
     )
-    task.log.info(f"resuming incremental replication", state)
+
+    sleep_for = timedelta()
+
+    if isinstance(state.cursor, datetime):
+        lag = datetime.now(tz=UTC) - state.cursor
+
+        if lag < binding.resourceConfig.interval:
+            sleep_for = binding.resourceConfig.interval - lag
+            task.log.info("incremental task ran recently, sleeping until `interval` has fully elapsed", {"sleep_for": sleep_for})
 
     while True:
-        # Yield to the event loop to prevent starvation.
-        await asyncio.sleep(0)
+        try:
+            if not task.stopping.event.is_set():
+                await asyncio.wait_for(
+                    task.stopping.event.wait(), timeout=sleep_for.total_seconds()
+                )
+
+            task.log.debug(f"incremental replication is idle and is yielding to stop")
+            return
+        except asyncio.TimeoutError:
+            pass  # `interval` elapsed.
 
         checkpoints = 0
         pending = False
+        
```

Comment:
this will log every iteration and be quite noisy on logs (hubspot drives this many times a second), instead of once at startup as it had been. Move outside of loop?

Reply:
moved